gcp/client.py

```diff
@@ -98,47 +186,28 @@ def _list(
         Internal function for calling .list() on some service's resource. Supports debug printing
         and caching of the response.
 
-        If you want empty call kwargs, pass in `{}`.
-        """
-        if call_kwargs is None:
-            call_kwargs = {"project": self.project_id}
-
-        ckey = cache_key(self.project_id, version, product, subproduct)
-        cached_result = self.cache.get(ckey, None)
-        if cached_result is not None:
-            print("found cached value for", ckey)
-            return cached_result
-
-        results = self._get_list_results(
-            product, subproduct, version, results_key, call_kwargs
-        )
-
-        if self.debug_cache:
-            print("setting cache value for", ckey)
-
-        self.cache.set(ckey, results)
-
-        return results
-
-    def _get_list_results(self, product, subproduct, version, results_key, call_kwargs):
-        """
-        Internal helper for actually constructing the .list() function call and calling it.
-
         If a service supports "zones", then loop through each zone to collect all resources from
         that service. An example of this is collecting all compute instances (compute.instances().list()).
         """
+
+        project_id = "-"
+        if "project" in call_kwargs:
+            project_id = call_kwargs["project"]
+
         service = self._service(product, version)
 
         api_entity = getattr(service, subproduct.split(".")[0])()
         for entity in subproduct.split(".")[1:]:
             api_entity = getattr(api_entity, entity)()
 
         if self.debug_calls:
-            print("calling", api_entity)
+            print(
```

Comment:
TODO: use the logging module and remove the random debug flags

as a new issue

Reply:
filed as https://github.com/mozilla/frost/issues/362